nidirect_contacts/nidirect_contacts.module

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * @file
+ * Contains nidirect_contacts.module.
+ */
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Link;
+use Drupal\views\ViewExecutable;
+use Drupal\views\Plugin\views\query\QueryPluginBase;
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function nidirect_contacts_form_views_exposed_form_alter(&$form, FormStateInterface $form_state) {
+  if ($form['#id'] == 'views-exposed-form-contacts-contact-search') {
+    // Make sort options invisible, but because the view embed filters are defined as exposed,
+    // we still get to use URL parameters despite having no visible form elements.
+    unset($form['sort_by']);
+    unset($form['sort_order']);
+  }
+
+  if ($form['#id'] == 'views-exposed-form-contacts-a-z-contacts-by-letter') {
+    $form['#action'] = \Drupal::service('router.route_provider')->getRouteByName('nidirect_contacts.default')->getPath();
+  }
+}
+
+/**
+ * Implements hook_views_query_alter().
+ */
+function nidirect_contacts_views_query_alter(ViewExecutable $view, QueryPluginBase $query) {
+  if ($view->id() == 'contacts_a_z' && $view->current_display == 'contacts_by_letter') {
+    // Argument condition referenced by array index feels a tad brittle. There
+    // doesn't seem to be a functional convenience function on the Views query object
+    // that lets us replace/alter the condition types and operators so it's set below.
+    $query->where[0]['conditions'][0]['value'] .= '%';
+    $query->where[0]['conditions'][0]['operator'] = 'LIKE';
+  }
+}
+
+/**
+ * Implements hook_preprocess_views_view().
+ */
+function nidirect_contacts_preprocess_views_view(&$variables) {
+  if (!preg_match('/^contacts/', $variables['id'])) {
+    return;
+  }
+
+  if (!in_array($variables['display_id'], ['contact_search', 'contacts_by_letter'])) {
+    return;
+  }
+
+  $q = \Drupal::request()->query->all();
```

Comment:
Not an issue, but there's the shorter Request class : https://api.drupal.org/api/drupal/vendor%21symfony%21http-foundation%21Request.php/class/Request/8.2.x

Reply:
Cool, thanks. The query parameters are likely accessible either way, I've followed what's typically done in core modules when in a .inc or .module context where they reference this static service.